x/auth/tx/service.go

```diff
@@ -161,9 +164,90 @@ func (s txServer) GetTx(ctx context.Context, req *txtypes.GetTxRequest) (*txtype
 	}, nil
 }
 
+// protoTxProvider is a type which can provide a proto transaction. It is a
+// workaround to get access to the wrapper TxBuilder's method GetProtoTx().
+// ref: https://github.com/cosmos/cosmos-sdk/issues/10347
+type protoTxProvider interface {
+	GetProtoTx() *txtypes.Tx
+}
+
 // GetBlockWithTxs returns a block with decoded txs.
 func (s txServer) GetBlockWithTxs(ctx context.Context, req *txtypes.GetBlockWithTxsRequest) (*txtypes.GetBlockWithTxsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "service not supported")
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "request cannot be nil")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+
+	if req.Height < 1 || req.Height > currentHeight {
+		return nil, sdkerrors.ErrInvalidHeight.Wrapf("requested height %d but height must not be less than 1 "+
+			"or greater than the current height %d", req.Height, currentHeight)
+	}
+
+	blockID, block, err := tmservice.GetProtoBlock(ctx, s.clientCtx, &req.Height)
+	if err != nil {
+		return nil, err
+	}
+
+	var offset, limit uint64
+	if req.Pagination != nil {
+		offset = req.Pagination.Offset
+		limit = req.Pagination.Limit
+	} else {
+		offset = 0
+		limit = pagination.DefaultLimit
+	}
+
+	blockTxs := block.Data.Txs
+	blockTxsLn := uint64(len(blockTxs))
+	txs := make([]*txtypes.Tx, 0, limit)
```

Comment:
`txs := make([]*txtypes.Tx, 0, limit)`
I wonder what's happening when the limit value is the max value of uint64.
Can it be an attack point to make the chain allocate lots of memory unnecessarily?

Reply:
Yes, it can be. And we proposed this problem to upstream (cosmos-sdk/issues/14917). So we will apply the changes later.